src/idle.rs

```diff
@@ -0,0 +1,200 @@
+                        let output = registry.bind::<wl_output::WlOutput, _, _>(name, 2, qh, ());
+                        state.outputs.insert(name, output);
+                    }
+                    "wl_seat" => {
+                        state.seat = Some(registry.bind::<wl_seat::WlSeat, _, _>(name, 7, qh, ()));
+                    }
+                    "zwlr_output_power_manager_v1" => {
+                        state.output_power_manager = Some(
+                            registry
+                                .bind::<zwlr_output_power_manager_v1::ZwlrOutputPowerManagerV1, _, _>(
+                                    name,
+                                    1,
+                                    qh,
+                                    (),
+                                ),
+                        );
+                    }
+                    "ext_idle_notifier_v1" => {
+                        state.idle_notifier = Some(
+                            registry.bind::<ext_idle_notifier_v1::ExtIdleNotifierV1, _, _>(
+                                name,
+                                1,
+                                qh,
+                                (),
+                            ),
+                        );
+                    }
+                    _ => return,
+                }
+                state.check_protocols_ready();
+            }
+            wl_registry::Event::GlobalRemove { name } => {
+                if let Some(output) = state.outputs.remove(&name) {
+                    output.release();
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+delegate_noop!(State: ignore wl_seat::WlSeat);
+delegate_noop!(State: ignore wl_output::WlOutput);
+delegate_noop!(State: ext_idle_notifier_v1::ExtIdleNotifierV1);
+delegate_noop!(State: zwlr_output_power_manager_v1::ZwlrOutputPowerManagerV1);
+delegate_noop!(State: ignore zwlr_output_power_v1::ZwlrOutputPowerV1);
+
+impl Dispatch<ext_idle_notification_v1::ExtIdleNotificationV1, ()> for State {
+    fn event(
+        state: &mut Self,
+        notification: &ext_idle_notification_v1::ExtIdleNotificationV1,
+        event: ext_idle_notification_v1::Event,
+        _: &(),
+        _: &Connection,
+        qh: &QueueHandle<Self>,
+    ) {
+        match event {
+            ext_idle_notification_v1::Event::Idled => {
+                if !state.is_idle {
+                    tracing::info!("System went idle, turning off displays");
+                    state.is_idle = true;
+                    state.set_display_power_mode(Mode::Off, qh);
+                }
+            }
+            ext_idle_notification_v1::Event::Resumed => {
+                if state.is_idle {
+                    tracing::info!("System resumed from idle, turning on displays");
+                    state.is_idle = false;
+                    state.set_display_power_mode(Mode::On, qh);
+                }
+                notification.destroy();
```

Comment:
Upon checking this in Cosmic, it does not seem necessary to destroy the notification every time a `Resumed` event is sent - the screen continues to turn off.

Is there some documentation page that mandates this?